src/box/lua/report_daemon.lua

```diff
@@ -0,0 +1,86 @@
+-- report_daemon.lua (internal file)
+
+local log   = require('log')
+local json  = require('json')
+local fiber = require('fiber')
+local http  = require('http.client')
+
+local PREFIX = "report_daemon"
+
+local daemon = {
+    reports_enabled  = false,
+    reports_interval = 0,
+    reports_host     = nil,
+    fiber            = nil,
+    control          = nil
+}
+
+local function fill_in_report(report)
+    report.tarantool_version = box.info.version
+    report.server_id         = box.info.uuid
+    report.cluster_id        = box.info.cluster.uuid
+    report.timestamp         = fiber.time64()
+    return report
+end
+
+local function version_report(self)
+    fiber.name(PREFIX, {truncate = true})
+    log.debug("version reports started")
+
+    local header = { report_type = "version", report_version = 1 }
+
+    while true do
+        local msg = self.control:get(self.reports_interval)
+        if msg == "reload" then
+            log.debug("version reports reloaded")
+        elseif msg == "shutdown" then
+            break
+        elseif msg == nil and box.info.status == "running" then
+            local ok, report = pcall(fill_in_report, header)
+            if ok then
+                http.post(self.reports_host, json.encode(report))
```

Comment:
Please double check that this doesn't block/slow things down too much.

Reply:
Considering we will have 1h as report_interval, there is no tcp_connection that lasts that long. Yes, there will be a handshake at each post, but it is a drop in the bucket